modules/openapi-generator/src/main/resources/typescript-axios/api.mustache

```diff
@@ -5,6 +5,14 @@
 {{^withSeparateModelsAndApi}}
 import { Configuration } from './configuration';
 import globalAxios, { AxiosPromise, AxiosInstance, AxiosRequestConfig } from 'axios';
+{{#withNodeImports}}
+// URLSearchParams not necessarily used
+// @ts-ignore
+import { URL, URLSearchParams } from 'url';
+{{#multipartFormData}}
+import FormData from 'form-data'
```

Comment:
Not sure if this should have `// @ts-ignore` – it is of course possible to set the `multipartFormData` parameter for a spec that does not have any form data endpoints and then this import would be unused (and probably the `form-data` dependency is missing too if not needed), but then again is it a user error if they configure to use something that is not needed?

Reply:
> is it a user error if they configure to use something that is not needed?

In my view, it totally is. Or otherwise put - if you declared your need for this code but it's not there (e.g. you want to manually extend the file), would that not be a generation error?

---

In my humble opinion, neither of `eslint-disable`, `tslint:disable` or `ts-ignore` should be present in this file. All of the unused imports can and should be fixed using `eslint --fix`, `prettier --write` or whatever code formatter someone uses. Since this is a programatic change, not a manual one, you can consistently apply it after each generation.

That's how I do it. I remove all these comments using `sed` after generating the source code and before formatting. The only warnings I get are related to the use of `any` typings. 

Besides, tslint has also been [deprecated since 2019](https://blog.palantir.com/tslint-in-2019-1a144c2317a9).

Btw, `typescript-node` has quite a few linting errors, including circular dependencies - yet another reason to deprecate that generator template.

Reply:
`eslint --fix` doesn't fix unused imports
and TS itself doesn't have autofix capability (so `noUnusedLocals` compiler setting will simply fail to compile the generated code)

Reply:
@amakhrov I'm using [eslint-plugin-unused-imports](https://www.npmjs.com/package/eslint-plugin-unused-imports) to remove unused imports. 

IMO, the generated code should either be valid, or not disable linting. The latter is especially tricky, since you must support all linting tooling (e.g. there should also be a `.prettierignore`), which to me seems unfeasable.

What other code generators do, is lint and/or format the code themselves after generation. That could be an option for JS as well.

But we are diverging from the topic of the PR.